core/io/read_spec.rb

```diff
@@ -122,13 +126,27 @@
   end
 
   it "reads only the specified number of bytes requested" do
-    IO.read("|sh -c 'echo hello'", 1).should == "h"
+    cmd = "|sh -c 'echo hello'"
+    platform_is :windows do
+      cmd = "|cmd.exe /C echo hello"
+    end
+    IO.read(cmd, 1).should == "h"
   end
 
   it "raises Errno::ESPIPE if passed an offset" do
-    lambda {
-      IO.read("|sh -c 'echo hello'", 1, 1)
-    }.should raise_error(Errno::ESPIPE)
+    platform_is_not :windows do
+      cmd = "|sh -c 'echo hello'"
+      lambda {
+        IO.read(cmd, 1, 1)
+      }.should raise_error(Errno::ESPIPE)
+    end
+    # TODO: https://bugs.ruby-lang.org/issues/12230
+    platform_is :windows do
+      cmd = "|cmd.exe /C echo hello"
+      lambda {
+        IO.read(cmd, 1, 1)
+      }.should raise_error(Errno::EINVAL)
+    end
```

Comment:
In that case where the difference is not only the `cmd`, the best is to split in two examples, and wrap these examples in platform guards.